internal/client/compose_test.go

```diff
@@ -0,0 +1,352 @@
+// Package client - compose_test contains functions to check the compose API
+// Copyright (C) 2020 by Red Hat, Inc.
+
+// Tests should be self-contained and not depend on the state of the server
+// They should use their own blueprints, not the default blueprints
+// They should not assume version numbers for packages will match
+// They should run tests that depend on previous results from the same function
+// not from other functions.
+//
+// NOTE: The compose fail/finish tests use fake composes so the following are not
+//       fully tested here:
+//
+//       * image download
+//       * log download
+//       * logs archive download
+//
+//       In addition osbuild-composer has not implemented:
+//
+//       * compose/results
+//       * compose/metadata
+package client
+
+import (
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/osbuild-composer/internal/weldr"
+)
+
+// Test the compose types API
+func TestComposeTypesV0(t *testing.T) {
+	composeTypes, resp, err := GetComposesTypesV0(testState.socket)
+	require.NoError(t, err, "failed with a client error")
+	require.Nil(t, resp)
+	require.Greater(t, len(composeTypes), 0)
+	var found bool
+	for _, t := range composeTypes {
+		if t.Name == "qcow2" && t.Enabled == true {
+			found = true
+			break
+		}
+	}
```

Comment:
How about the scenario where all of the enabled compose types are hard-coded as expected values ? 

Or do we not want to do this anymore ? Can still do it on the composer-cli side though (it's actually already there, just need to switch the backend rpm and possibly leave only the cli tests, not the image building ones).

Reply:
I think that's better on the cli side, I wanted to make sure these run against both backends, and as unit tests, without a bunch of 'which test am I running' logic embedded in it. FYI the unit test only has qcow2 mocked.